lib/Sema/CSDiag.cpp

```diff
@@ -2477,6 +2477,14 @@ diagnoseUnviableLookupResults(MemberLookupResult &result, Type baseObjTy,
           diagnose(loc, diag::did_you_mean_raw_type);
           return; // Always prefer this over typo corrections.
         }
+      } else if (baseObjTy->isAny()) {
+        if (auto DRE = dyn_cast<DeclRefExpr>(baseExpr)) {
+          auto name = DRE->getDecl()->getName().str().str();
+          std::string replacement = "(" + name + " as AnyObject)";
```

Comment:
@jrose-apple I've pushed a PR for master. Please take a look and I'll put up a PR for swift-3.0-branch if it tests and merges successfully.

https://github.com/apple/swift/pull/4116